adsft/extraction.py

```diff
@@ -608,29 +608,29 @@ def parse_xml(self, preferred_parser_names=None):
 
         for parser_name in preferred_parser_names:
             parsed_xml = self._parse_xml(parser_name)
+
             logger.debug("Checking if the parser '{}' succeeded".format(parser_name))
             success = False
-            for xpath in META_CONTENT[self.meta_name].get('fulltext', {}).get('xpath', []):
-                fulltext = None
-                fulltext_elements = parsed_xml.xpath(xpath)
-                if len(fulltext_elements) > 0:
-                    fulltext = u" ".join(map(unicode.strip, map(unicode, fulltext_elements[0].itertext())))
-                    fulltext = TextCleaner(text=fulltext).run(decode=False, translate=True, normalise=True, trim=True)
-                if not fulltext:
-                    continue
-                else:
-                    logger.debug("The parser '{}' succeeded".format(parser_name))
-                    success = True
+            ft_found = False
+            for content_name in META_CONTENT[self.meta_name]:
+                if ft_found:
                     break
-
-            if not success:
-                logger.debug("The parser '{}' failed".format(parser_name))
-            else:
+                for xpath in META_CONTENT[self.meta_name].get(content_name, {}).get('xpath', []):
+                    element_found = False
+                    elements = parsed_xml.xpath(xpath)
+                    if len(elements) > 0:
+                        element_found = True
+                        success = True
+                    if content_name == 'fulltext' and element_found:
+                        ft_found = True
+                        fulltext = u" ".join(map(unicode.strip, map(unicode, elements[0].itertext())))
+                        fulltext = TextCleaner(text=fulltext).run(decode=False, translate=True, normalise=True, trim=True)

```

Comment:
The `fulltext` variable is never used after these two lines, in the previous version of the code that was used to decide that the full text extraction was successful. Either we remove these two lines or we use the outcome with something like:
```
if fulltext:
    ft_found = True
```
And we remove the `ft_found = True` in line 625. But if we do the latter, then when we get legit empty bodies, the code will go ahead and try the next parser. That's what we were trying to avoid, so we could change the condition to:
```
if isinstance(fulltext, str):
    ft_found = True
```
Does this sound reasonable?

Reply:
Yes it sounds reasonable. I'm leaning towards removing the two lines unless you say otherwise.